src/auth/authclient.ts

```diff
@@ -28,5 +34,5 @@ export abstract class AuthClient {
    * Provides an alternative request
    * implementations with auth credentials.
    */
-  abstract request(opts: request.Options): request.Request|void;
+  abstract request(opts: AxiosRequestConfig): Promise<BodyResponse>;
```

Comment:
This is a breaking change, but keeping the function name the same would create a landmine for users. Some alternatives:

* Does this need to be public API at all?
* Could we provide an API compatible implementation that provides `request.Result` on top of axios (i.e. avoid this being a semver major).
* If semver major is unavoidable and we need to expose this API, can we rename it to make the migration for dependents easier?

Reply:
I spot checked a few dependents, and it looks like folks aren't directly using this method (though it's always hard to say).   A few thoughts:
- We could try making this protected, and see what happens :)
- I don't think wrapping this with request style APIs would be successful.  There are way too many differences between Axios and Request, and I don't want to manage a janky compat shim layer if possible.
- Another option here is to check for common params that could come in with a request Options object that aren't present on an Axios config, and throw if I see them with a humane warning.

I would prefer either making this protected, and/or more humane defensive parameter checking with appropriate exceptions. 

Eh?